src/app/app.component.spec.ts

```diff
@@ -3,15 +3,137 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-import { TestBed } from '@angular/core/testing';
-import { AppComponent } from './app.component';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing'
+import { By } from '@angular/platform-browser'
+import { AppComponent } from './app.component'
 
 describe('AppComponent', () => {
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       declarations: [
         AppComponent
       ],
-    }).compileComponents();
+    }).compileComponents()
   });
-});
+
+  const TRIPLE_BANANA_SHARE_URL = "https://triplebanana.dev/share"
+
+  const USER_AGENT_ANDROID_CHROME_BROWSER =
+    "Mozilla/5.0 (Linux; Android 12; SM-G996N) \
+    AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Mobile Safari/537.36"
+
+  const USER_AGENT_ANDROID_KAKAOTALK_WEB_VIEW =
+    "Mozilla/5.0 (Android; Mobile; rv:13.0) \
+    Gecko/13.0 Firefox/13.0 KAKAOTALK"
+
+  const USER_AGENT_LINUX_CHROME =
+    "Mozilla/5.0 (X11; Linux x86_64) \
+    AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.61 Safari/537.36"
+
+  it('should move android intent including target url when use chrome browser', fakeAsync(() => {
+    // given
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+
+    const shareUrl = new URL("https://google.com")
+    spyOn<any>(app, 'getCurrentUrl').and.returnValue(generateShareLink(shareUrl.toString()))
```

Comment:
As we already discussed in Google Meet, I don't think private API stubbing is a good way. The `getCurrentUrl()` is private API. The private API might be changed anytime. The test would faile as the internal implementation changes by refactoring. If we rewrite/modify tests whenever our implementation is changed, the overhead of writing the tests would be huge and we can't trust the tests.